datafusion/proto/src/logical_plan/from_proto.rs

```diff
@@ -999,7 +1002,7 @@ pub fn parse_expr(
             let operands = binary_expr
                 .operands
                 .iter()
-                .map(|expr| parse_expr(expr, registry))
+                .map(|expr| parse_expr(expr, registry, codec))
```

Comment:
In from_proto.rs file, a lot of places need to change like this line.
I haven't finish it but submit this draft PR to verify the road.